src/PowerShell/Microsoft.WinGet.DSC/Microsoft.WinGet.DSC.psm1

```diff
@@ -193,130 +187,169 @@ class WinGetAdminSettings
-    [WinGetAction]$Action = [WinGetAction]::Full
+    [WinGetEnsure]$Ensure = [WinGetEnsure]::Present
 
-    # Gets the current sources on winget.
-    [WinGetSources] Get()
+    [WinGetSource] Get()
     {
-        Assert-WinGetCommand "Get-WinGetSource"
-        $packageCatalogReferences = Get-WinGetSource
-        $wingetSources = [List[Hashtable]]::new()
-        foreach ($packageCatalogReference in $packageCatalogReferences)
-        {
-            $source = @{
-                $packageCatalogReference.Name = @{
-                    Identifier = $packageCatalogReference.Id
-                    Arg = $packageCatalogReference.Argument
-                    Type = $packageCatalogReference.Type
-                }
-            }
-            $wingetSources.Add($source)
+        $currentSource = $null
+
+        try {
+            $currentSource = Get-WinGetSource -Name $this.Name
+        }
+        catch {
         }
 
-        $result = @{
-            SID = ''
-            Sources = $wingetSources
+        $result = [WinGetSource]::new()
+
+        if ($currentSource)
+        {
+            $result.Ensure = [WinGetEnsure]::Present
+            $result.Name = $currentSource.Name
+            $result.Argument = $currentSource.Argument
+            $result.Type = $currentSource.Type
+            $result.TrustLevel = $currentSource.TrustLevel
+            $result.Explicit = $currentSource.Explicit
+        }
+        else
+        {
+            $result.Ensure = [WinGetEnsure]::Absent
+            $result.Name = $this.Name
         }
+
         return $result
     }
 
-    # Tests if desired properties match.
     [bool] Test()
     {
-        $currentSources = $this.Get().Sources
-        $currentState = [Ensure]::Present
+        return $this.TestAgainstCurrent($this.Get())
+    }
+
+    [void] Set()
+    {
+        Assert-IsAdministrator
+
+        $currentSource = $this.Get()
 
-        # If this is a full match and the counts are different return false. This only applies if we want to ensure the full source is present.
-        if (($this.Action -eq [WinGetAction]::Full) -and ($this.Sources.Count -ne $currentSources.Count) -and ($this.Ensure -eq [Ensure]::Present))
+        $removeSource = $false
+        $resetSource = $false
+        $addSource = $false
+
+        if ($this.Ensure -eq [WinGetEnsure]::Present)
         {
-            return $false
+            if ($currentSource.Ensure -eq [WinGetEnsure]::Present)
+            {
+                if (-not $this.TestAgainstCurrent($currentSource))
+                {
+                    $resetSource = $true
+                    $addSource = $true
+                }
+                # else in desired state
+            }
+            else
+            {
+                $addSource = $true
+            }
+        }
+        else
+        {
+            if ($currentSource.Ensure -eq [WinGetEnsure]::Present)
+            {
+                $removeSource = $true
+            }
+            else

```

Comment:
I think I like more the else comment like 265, but at least both should be consistent